tests/unit/db/design_document_tests.py

```diff
@@ -164,35 +213,261 @@ def test_delete_a_view(self):
         ddoc.delete_view('view001')
         self.assertEqual(ddoc.get('views'), {})
 
-    def test_fetch(self):
+    def test_delete_a_query_index_view(self):
+        """
+        Test deleting a query index view fails as expected.
+        """
+        # This is not the preferred way of dealing with query index
+        # views but it works best for this test.
+        data = {
+            '_id': '_design/ddoc001',
+            'language': 'query',
+            'views': {
+                'view001': {'map': {'fields': {'name': 'asc', 'age': 'asc'}},
+                            'reduce': '_count',
+                            'options': {'def': {'fields': ['name', 'age']},
+                                        'w': 2}
+                            }
+                    }
+        }
+        self.db.create_document(data)
+        ddoc = DesignDocument(self.db, '_design/ddoc001')
+        ddoc.fetch()
+        try:
+            ddoc.delete_view('view001')
```

Comment:
I replaced every instance of exception checking with assertRaises in the code that was added/updated in PR #75, PR #77 (this PR), and PR #79.  I also created Issue #80 so that we can circle back and make this change in all of our unit tests.